shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -2269,25 +2274,23 @@ - (NSString*)contrastMode {
 #pragma mark - Status bar style
 
 - (UIStatusBarStyle)preferredStatusBarStyle {
-  return _statusBarStyle;
+  return self.statusBarStyle;
 }
 
 - (void)onPreferredStatusBarStyleUpdated:(NSNotification*)notification {
   // Notifications may not be on the iOS UI thread
+  __weak FlutterViewController* weakSelf = self;
   dispatch_async(dispatch_get_main_queue(), ^{
     NSDictionary* info = notification.userInfo;
-
     NSNumber* update = info[@(flutter::kOverlayStyleUpdateNotificationKey)];
-
     if (update == nil) {
       return;
     }
 
-    NSInteger style = update.integerValue;
-
-    if (style != _statusBarStyle) {
-      _statusBarStyle = static_cast<UIStatusBarStyle>(style);
-      [self setNeedsStatusBarAppearanceUpdate];
+    UIStatusBarStyle style = static_cast<UIStatusBarStyle>(update.integerValue);
+    if (style != weakSelf.statusBarStyle) {
```

Comment:
This is using self more than once, so should get a `strongSelf` dance or extraction to a helper method that takes `style` as a param.

(Probably doesn't really matter in this case, but I don't see a good reason not to follow preferred patterns here.)

Reply:
Good catch. Done.